src/hwacc/HWModeling/src/cycle_counts.cc

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2025 Akanksha Chaudhari, Matt Sinclair
+ * All rights reserved.
+ *
+ * This file contains modifications and/or code derived from:
```

Comment:
@powerjg will have to chime in, but I'm not sure we can put this in the license?